src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,63 @@
+import { useSearchParams } from 'react-router-dom';
+import { SearchLink } from './SearchLink';
+import classNames from 'classnames';
+import React from 'react';
+import { getSearchWith } from '../utils/searchHelper';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || null;
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  const availableCenturies = ['16', '17', '18', '19', '20'];
+
+  const getCenturiesParams = (century: string) => {
+    const preparedCenturies = centuries.includes(century)
+      ? centuries.filter(cent => cent !== century)
+      : [...centuries, century];
+
+    return { centuries: preparedCenturies };
+  };
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchParams(
+      getSearchWith(searchParams, {
+        query: `${event.target.value.toLowerCase()}` || null,
+      }),
+    );
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
+        <SearchLink
+          className={classNames({
+            'is-active': !sex,
+          })}
+          params={{ sex: null }}
+        >
           All
-        </a>
-        <a className="" href="#/people?sex=m">
+        </SearchLink>
+
+        <SearchLink
+          className={classNames({
+            'is-active': sex === 'm',
```

Comment:
Create a enum for 'm', 'f' and use it everywhere

Reply:
Done.